qiita_db/study.py

```diff
@@ -347,6 +348,59 @@ def create(cls, owner, title, efo, info, investigation=None):
 
         return cls(study_id)
 
+    @classmethod
+    def delete(cls, id_):
+        r"""Deletes the study from the database
+
+        Parameters
+        ----------
+        id_ : integer
+            The object identifier
+
+        Raises
+        ------
+        QiitaDBError
+            If the sample_(id_) table exists means a sample template exists
+        """
+        cls._check_subclass()
+
+        # checking that the id_ exists
+        cls(id_)
```

Comment:
No, but you are only instantiating the object to check if the `id` exists. I think that is unnecessary given that we have the function `_check_id`. i.e. `_check_id` is called in `__init__` (among other things) so you avoid all these other things by using the function that you really want.